src/common/options.cc

```diff
@@ -4430,6 +4430,20 @@ std::vector<Option> get_global_options() {
 
     Option("bluestore_rocksdb_cf", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
     .set_default(true)
+#ifdef WITH_SEASTAR
+    // This is necessary as the Seastar's allocator imposes restrictions
+    // on the number of threads that entered malloc/free/*. Unfortunately,
+    // RocksDB sharding in BlueStore dramatically lifted the number of
+    // threads spawn during RocksDB's init.
+    .set_validator([](std::string *value, std::string *error_message){
+      if (*value != "false") {
```

Comment:
i am okay with this. but please note, in `Option::parse_value()`, we interpret non-zero numbers as true.

Reply:
It's even more problematic due to the case sensitivity:

```
  } else if (type == Option::TYPE_BOOL) {
    if (strcasecmp(val.c_str(), "false") == 0) {
      *out = false;
    } else if (strcasecmp(val.c_str(), "true") == 0) {
      *out = true;
    } else { 
      int b = strict_strtol(val.c_str(), 10, error_message);
      if (!error_message->empty()) {
        return -EINVAL;
      }
      *out = (bool)!!b;
    }
```

Looks the config subsystem should get something similar to `strict_si_cast()` but for boolean.

Reply:
yeah, something like
```c++
template<typename T>
T parse(std::string_view s)
{
  // magic here
  return parser_for_t(s);
}
```